### ISSUE TITLE
Setting pygame.SRCALPHA influences rotation outcome

### ISSUE THREAD

**Issue by SteveClement** [pygame/pygame#4298]
**Environment:**

- **Operating system** : MacOS but also tested on Linux
- **Python version** : 3.11.1
- **SDL version** : 2.28.4
- **PyGame version** : 2.6.0

**Current behavior:**

The following script rotates a red square around itself: 

https://gist.github.com/SteveClement/5997981ee65fa68eacc7c886a0f5b2a7

Removing pygame.SRCALPHA from that script stops rotating it and instead grows and shrinks it.

The same without SRCALPHA: https://gist.github.com/SteveClement/8d578802feda90e310d46d33ac6b03c8

I am not sure whats happening and if this is a legit but or ChatGPT generating pooPoo code... ;)

```
diff -u rotate-square.py weird-square.py
--- rotate-square.py	2024-07-18 09:55:44
+++ weird-square.py	2024-07-18 10:03:10
@@ -9,7 +9,7 @@
 pygame.display.set_caption('Rotate Surface on Its Own Axis')

 # Create a surface (a red square for demonstration)
-surface = pygame.Surface((50, 50), pygame.SRCALPHA)
+surface = pygame.Surface((50, 50))
 surface.fill((255, 0, 0))

 # Initial position
@@ -52,4 +52,3 @@
 # Clean up
 pygame.quit()
 sys.exit()
```

**Comment by SteveClement** [comment]
Reproduced with:

pygame 2.6.1.dev1 (SDL 2.30.5, Python 3.9.6)

**Comment by Vyomm03** [comment]
Yes, experiencing the same issue. It took me hours to figure out why my rectangle was not rotating along its center.
